### PR TITLE
Fix vagrant builds

### PR DESCRIPTION
**Pull request by dwoz**
This PR fixes vagrant builds for salt-jenkins allowing converge to be run against a fresh windows image. Salt kitchen installs a minion from the windows installer. We then use salt-jenkins to bootstrap the Winrepo package repository. Having Winrepo available allows us to install git, a Microsoft compiler and a python to test against. Including the `windeps` state before git.salt will bootstrap the system.

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -169,4 +175,4 @@ suites:
           test_git_commit: <%= ENV['TRAVIS_BRANCH'] == 'master' ? 'develop' : ENV['TRAVIS_BRANCH'] %>
           py3: true
         windows.sls:
-          virtualenv_path: 'c:\Python36\Scripts\pip.exe'
+          virtualenv_path: 'c:\Program Files (x86)\Python35-32\Scripts\pip.exe'
```
Comment (gtmanfred) [inline .kitchen.yml]: does the Python27 also need to be fixed, or are we just using the python that comes with salt?

File: git/salt.sls [hunk 2]
```diff
@@ -204,8 +204,16 @@ include:
...
+testing-dir:
+  file.directory:
+    - name: {{ testing_dir }}
+  {%- if grains['os'] == 'Windows' %}
+    - win_owner: 'Users'
+    - win_inheritance: true
+    - win_perms:
+      Users:
+        perms: full_control
+  {%- endif %}
```
Comment (gtmanfred) [inline git/salt.sls]: nice

File: python/dulwich.sls [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+dulwich:
+  pip.installed:
+    - name: dulwich
+    - bin_env: {{ salt['config.get']('virtualenv_path', '') }}
+    - cwd: {{ salt['config.get']('pip_cwd', '') }}
+{% if grains['os'] not in ('Windows',) %}
+    - require:
+      - cmd: pip-install
+{% endif %}
```
Comment (gtmanfred) [inline python/dulwich.sls]: This feels gross, but I understand why we are adding it. so :+1:

File: .kitchen.yml (hunk 1 above)
Comment (dwoz) [reply .kitchen.yml]: The python2 path was already pointing to the correct place (where python2 is getting installed by windeps).

File: python/dulwich.sls (hunk 3 above)
Comment (dwoz) [reply python/dulwich.sls]: Yes, it is maybe not the greatest. It'll do the job for now though.

File: .kitchen.yml [hunk 4]
```diff
@@ -156,6 +159,9 @@ suites:
       - centos-6
       - ubuntu-1404
     provisioner:
+      salt_minion_extra_config:
+        winrepo_remotes_ng:
+          - https://github.com/dwoz/salt-winrepo-ng.git
```
Comment (dwoz) [inline .kitchen.yml]: This should be temporary. Maybe we can get https://github.com/saltstack/salt-winrepo-ng/pull/1333 merged and then remove it.

File: git/salt.sls [hunk 5]
```diff
@@ -226,7 +234,7 @@ clone-salt-repo:
       {%- endif %}
       - file: /usr/bin/busybox
       {%- endif %}
-      - file: {{ testing_dir }}
+      - testing-dir
```
Comment (twangboy) [inline git/salt.sls]: Shouldn't this still be `- file: testing-dir`?

File: python/psutil.sls [hunk 6]
```diff
@@ -11,10 +11,9 @@ include:
...
     - upgrade: True
-    {%- if salt['config.get']('virtualenv_path', None) %}
-    - bin_env: {{ salt['config.get']('virtualenv_path') }}
-    {%- endif %}
-    {%- if grains['os'] not in ('Windows',) %}
+    - bin_env: {{ salt['config.get']('virtualenv_path', '') }}
+    - cwd: {{ salt['config.get']('pip_cwd', '') }}
+    {%- if grains['os'] not in 'Windows' %}
     - require:
       {%- if grains['os_family'] not in ('Arch', 'Solaris', 'FreeBSD', 'Gentoo', 'MacOS', 'Windows') %}
```
Comment (twangboy) [inline python/psutil.sls]: You could probably remove Windows from this list since we're gating it above.

Comment (dwoz) [reply python/psutil.sls]: Ack, that buggy too. `in 'Windows'` not `in ('Windows',)`

File: git/salt.sls (hunk 5 above)
Comment (gtmanfred) [reply git/salt.sls]: It should be, but it is not required, you can now just include the stateid name

@dwoz can we get this one fixed, just so we are consistent everywhere?

Comment (twangboy) [reply git/salt.sls]: I always thought it should be that way... I hated putting the module in front... Glad they added that.

Comment (dwoz) [reply git/salt.sls]: Fixed.

**Comment by dwoz** [comment]
@gtmanfred I've picked through this and knocked off the rough edges. All the tests are passing again after pinning pip to 9.0.1. A new version 9.0.2 was released yesterday and it broke a lot of the circle ci builds. see [this pip issue](https://github.com/pypa/pip/issues/1087)

**Pull request merged**